src/code/z_scene.c

```diff
@@ -3,116 +3,135 @@
+    ASSERT(((objectCtx->numEntries < ARRAY_COUNT(objectCtx->slots)) &&
+            (((s32)objectCtx->slots[objectCtx->numEntries].segment + size) < (u32)objectCtx->spaceEnd)),
            "this->num < OBJECT_EXCHANGE_BANK_MAX && (this->status[this->num].Segment + size) < this->endSegment",
            "../z_scene.c", 142);
 
-    DmaMgr_SendRequest1(objectCtx->status[objectCtx->num].segment, gObjectTable[objectId].vromStart, size,
+    DmaMgr_SendRequest1(objectCtx->slots[objectCtx->numEntries].segment, gObjectTable[objectId].vromStart, size,
                         "../z_scene.c", 145);
 
-    if (objectCtx->num < OBJECT_EXCHANGE_BANK_MAX - 1) {
-        objectCtx->status[objectCtx->num + 1].segment =
-            (void*)ALIGN16((s32)objectCtx->status[objectCtx->num].segment + size);
+    if (objectCtx->numEntries < (ARRAY_COUNT(objectCtx->slots) - 1)) {
+        objectCtx->slots[objectCtx->numEntries + 1].segment =
+            (void*)ALIGN16((s32)objectCtx->slots[objectCtx->numEntries].segment + size);
     }
 
-    objectCtx->num++;
-    objectCtx->unk_09 = objectCtx->num;
+    objectCtx->numEntries++;
+    objectCtx->numPersistentEntries = objectCtx->numEntries;
 
-    return objectCtx->num - 1;
+    return objectCtx->numEntries - 1;
 }
 
-void Object_InitBank(PlayState* play, ObjectContext* objectCtx) {
-    PlayState* play2 = play; // Needs to be a new variable to match (possibly a sub struct?)
+// 1000 kilobytes
+#define OBJECT_SPACE_SIZE_BASE (1000 * 1024)
+
+void Object_InitContext(PlayState* play, ObjectContext* objectCtx) {
+    PlayState* play2 = play;
     u32 spaceSize;
     s32 i;
 
     if (play2->sceneNum == SCENE_SPOT00) {
-        spaceSize = 1024000;
+        spaceSize = OBJECT_SPACE_SIZE_BASE;
     } else if (play2->sceneNum == SCENE_GANON_DEMO) {
         if (gSaveContext.sceneSetupIndex != 4) {
-            spaceSize = 1177600;
+            spaceSize = OBJECT_SPACE_SIZE_BASE + (150 * 1024);
         } else {
-            spaceSize = 1024000;
+            spaceSize = OBJECT_SPACE_SIZE_BASE;
         }
     } else if (play2->sceneNum == SCENE_JYASINBOSS) {
-        spaceSize = 1075200;
+        spaceSize = OBJECT_SPACE_SIZE_BASE + (50 * 1024);
     } else if (play2->sceneNum == SCENE_KENJYANOMA) {
-        spaceSize = 1075200;
+        spaceSize = OBJECT_SPACE_SIZE_BASE + (50 * 1024);
     } else if (play2->sceneNum == SCENE_GANON_BOSS) {
-        spaceSize = 1075200;
+        spaceSize = OBJECT_SPACE_SIZE_BASE + (50 * 1024);
     } else {
-        spaceSize = 1024000;
+        spaceSize = OBJECT_SPACE_SIZE_BASE;
     }
 
-    objectCtx->num = objectCtx->unk_09 = 0;
-    objectCtx->mainKeepIndex = objectCtx->subKeepIndex = 0;
+    objectCtx->numEntries = objectCtx->numPersistentEntries = 0;
+    objectCtx->mainKeepSlot = objectCtx->subKeepSlot = 0;
 
-    for (i = 0; i < OBJECT_EXCHANGE_BANK_MAX; i++) {
-        objectCtx->status[i].id = OBJECT_INVALID;
+    for (i = 0; i < ARRAY_COUNT(objectCtx->slots); i++) {
```

Comment:
I don't really see a reason it should be a define tbh, besides the debug string saying it originally was

I also think the code reads better with array_count since you already gotta watch out for iterations from / up to `numEntries`/`numPersistentEntries`, no need to add another name imo?

Reply:
Personally I prefer the ARRAY_COUNT solution instead of making extra defines for this kind of stuff.
It may be worth noting somewhere this was a macro and the original name (as a way of preserving history or something idk), probably in the struct itself